### DISCUSSION TITLE
Setting `--spi` provider args?

### DISCUSSION THREAD

**Discussion by xgp** [dasniko/testcontainers-keycloak#110] [Q&A] [answered]
I'd like to set provider args, e.g. `--spi-theme-cache-themes=false` or `--spi-email-template-provider=mycustomprovider`. Is there a way to do that using the `KeycloakContainer` fluent api?

**Comment by xgp** [comment, marked as answer]
I believe you can use the convention of upper-casing and adding a `KC` prefix as an env var. E.g. `--spi-email-template-provider` becomes `KC_SPI_EMAIL_TEMPLATE_PROVIDER` and you can use the `withEnv("KC_SPI_EMAIL_TEMPLATE_PROVIDER", "mycustomprovider")` method.

Also, regarding the specific need to disable theme caching, looks like there is a `withDisabledCaching()` method.

**Reply by dasniko** [reply]
You are completely right in your assumptions 🙂
